src/app/services/comment-period.service.ts

```diff
@@ -0,0 +1,115 @@
+import { Injectable } from '@angular/core';
+import { Response } from '@angular/http';
+import { CommentPeriod } from '../models/commentperiod';
+import { Comment } from '../models/comment';
+import { Document } from '../models/document';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/switchMap';
+
+import { Api } from './api';
+import { ValuedComponent } from '../models/vcs';
+
+@Injectable()
+export class CommentPeriodService {
+  pcp: CommentPeriod;
+
+  constructor(private api: Api) { }
+
+  // return a public comment period object
+  getByCode(id: string, code: string): Observable<CommentPeriod> {
+  // Grab the project data first
+    return this.api.getPCPByCode(id)
+      .map((res: Response) => res.json())
+      .map((pcp: CommentPeriod) => {
+        if (!pcp) {
+          return Observable.throw(new Error('PCP not found'));
+        }
+
+        this.pcp = pcp;
```

Comment:
Should assign proper instance of CommentPeriod class, i.e.:

```
this.pcp = new CommentPeriod(pcp);
```

Reply:
Done